tests/integration/test_distributed_config/test.py

```diff
@@ -0,0 +1,49 @@
+import pytest
+from helpers.cluster import ClickHouseCluster
+import logging
+
+cluster = ClickHouseCluster(__file__)
+node = cluster.add_instance("node", main_configs=["configs/overrides.xml"])
+
+
+@pytest.fixture(scope="module")
+def start_cluster():
+    try:
+        logging.info("Starting cluster...")
+        cluster.start()
+        logging.info("Cluster started")
+
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def test_distibuted_settings(start_cluster):
+    node.query("")
+    node.query(
+        """
+        CREATE TABLE data_1 (key Int) ENGINE Memory();
+        CREATE TABLE dist_1 as data_1 ENGINE Distributed(default, default, data_1) SETTINGS flush_on_detach = true;
+        SYSTEM STOP DISTRIBUTED SENDS dist_1;
+        INSERT INTO dist_1 SETTINGS prefer_localhost_replica=0 VALUES (1);
+        DETACH TABLE dist_1;
+    """
+    )
+    assert "flush_on_detach = 1" in node.query("SHOW CREATE dist_1")
+    # flush_on_detach=true, so data_1 should have 1 row
+    assert int(node.query("SELECT count() FROM data_1")) == 1
+
+    node.query(
+        """
+        CREATE TABLE data_2 (key Int) ENGINE Memory();
+        CREATE TABLE dist_2 as data_2 ENGINE Distributed(default, default, data_2);
+        SYSTEM STOP DISTRIBUTED SENDS dist_2;
+        INSERT INTO dist_2 SETTINGS prefer_localhost_replica=0 VALUES (2);
+        DETACH TABLE dist_2;
+    """
+    )
+    ## Settings are not added to CREATE (only specific one, like index_granularity for MergeTree)
+    # assert "flush_on_detach = 0" in node.query("SHOW CREATE dist_2")
+
+    # But settins are applied (flush_on_detach=false in config, so data_2 should not have any rows)
+    assert int(node.query("SELECT count() FROM data_2")) == 0
```

Comment:
I guess `flush_on_detach = false` only means that flush is not mandatory upon detach but the database could flush the data regardless (not 100% sure).  Let's check if the test is flaky.

Reply:
(unfortunately the other Distributed table engine options make it difficult to write a proper test)

Reply:
>I guess flush_on_detach = false only means that flush is not mandatory upon detach but the database could flush the data regardless (not 100% sure).

No. If flush_on_detach=false, then the flush will not happens, otherwise will. But I guess you already figure this out (according to your patch into the docs - https://github.com/ClickHouse/ClickHouse/pull/59291/commits/1751d7623457b16a67e9762e7ca3b64b2f6c19c1)

> Let's check if the test is flaky.

AFAICS the flaky check passed - https://s3.amazonaws.com/clickhouse-test-reports/59291/4248ae509a5b6ce7debe6a994b2c238cb1df2c96/integration_tests_flaky_check__asan_.html